python/taichi/lang/ir_builder.py

```diff
@@ -146,7 +161,31 @@ def build_Call(ctx, node):
             else:
                 args.append(arg.ptr)
         keywords = dict(ChainMap(*[keyword.ptr for keyword in node.keywords]))
-        node.ptr = node.func.ptr(*args, **keywords)
+
+        if isinstance(node.func, ast.Attribute):
+            attr_name = node.func.attr
+            if attr_name == 'format':
```

Comment:
Should we check if node.func.value.ptr is str?

Reply:
Make sense

Reply:
But what else could it be?

Reply:
Like an instance of a class with a method named "format"?